sdk/core/azure-core/src/main/java/com/azure/core/util/polling/OperationResourcePollingStrategy.java

```diff
@@ -149,6 +151,9 @@ public Mono<PollResponse<T>> onInitialResponse(Response<?> response, PollingCont
             Duration retryAfter = ImplUtils.getRetryAfterFromHeaders(response.getHeaders(), OffsetDateTime::now);
             return PollingUtils.convertResponse(response.getValue(), serializer, pollResponseType)
```

Comment:
PS: The impl of `convertResponse` is somehow problematic.
https://github.com/Azure/azure-sdk-for-java/blob/f9246a09f1f48dca69fa582dba1bdba00c91269b/sdk/core/azure-core/src/main/java/com/azure/core/util/polling/implementation/PollingUtils.java#L114-L115

It likely won't work well, if JSON contains some read-only properties.

Reply:
issue https://github.com/Azure/azure-sdk-for-java/issues/34248